eldritch/mythos.py

```diff
@@ -412,6 +422,163 @@ def get_trigger(self, event, state):
     return None
 
 
+class Mythos16(Environment):
+  def __init__(self):
+    super().__init__(
+        "Mythos16", "WitchHouse", "Cave",
+        {"square", "diamond"}, {"circle"},
+        "urban", activity_location="Uptown"
+    )
+
+  def get_trigger(self, event, state):
+    if not isinstance(event, events.Movement):
+      return None
+    if event.character.place.name != self.activity_location:
+      return None
+    return events.BinaryChoice(
+        event.character,
+        "Shop at estate sale?", "Yes", "No",
+        events.Purchase(event.character, "unique", draw_count=2, keep_count=2),
+        events.Nothing(),
+    )
+
+
+class Mythos17(Environment):
+  def __init__(self):
+    super().__init__(
+        "Mythos17", "WitchHouse", "Cave",
+        {"plus"}, {"moon"},
+        "urban", activity_location="University"
+    )
+
+  def get_trigger(self, event, state):
+    if not isinstance(event, events.Movement):
+      return None
+    if event.character.place.name != self.activity_location:
+      return None
+    check = events.Check(event.character, "lore", -1)
+    gain = events.Gain(event.character, {"clues": 1})
+    return events.PassFail(event.character, check, gain, events.Nothing())
+
+
+class Mythos18(Environment):
+  def __init__(self):
+    super().__init__("Mythos18", "Square", "Unnamable", {"plus"}, {"moon"}, "mystic")
+
+  def get_interrupt(self, event, state):
+    if (
+        isinstance(event, events.GainOrLoss)
+        and values.Calculation(event.gains["sanity"], operand=values.operator.gt, right=0).value(
+            state)
+        and (event.source is None or event.source.name not in ("Psychology", "Asylum"))
+    ):
+      return events.GainPrevention(self, event, "sanity", event.gains["sanity"])
+    return None
+
+
+class Mythos19(Headline):
```

Comment:
Can you also add tests for this? Maybe stick a monster and a character in the merchant streets. Validate that this card gets removed at the end of the next mythos?